ctera/src/main/java/ch/cyberduck/core/ctera/CteraSession.java

```diff
@@ -154,14 +220,42 @@ protected void logout() throws BackgroundException {
     @Override
     @SuppressWarnings("unchecked")
     public <T> T _getFeature(final Class<T> type) {
+        final CteraSession session = this;
         if(type == Touch.class) {
-            return (T) new CteraTouchFeature(this);
+            return (T) new CteraTouchFeature(this) {
+                @Override
+                public void preflight(final Path workdir, final String filename) throws BackgroundException {
+                    // TODO CTERA-136 do we need to require writepermission as well?
+                    super.preflight(workdir, filename);
+                    session.checkCteraRole(workdir, CREATEFILEPERMISSION);
+                }
+            };
         }
         if(type == Directory.class) {
-            return (T) new CteraDirectoryFeature(this);
+            return (T) new CteraDirectoryFeature(this) {
```

Comment:
Extract inner class to upper.

Reply:
Done e57ba170367804bdddabca3d79878d9cc6a410ec